### PR TITLE
rust: update to 1.36.0

### PR DESCRIPTION
**Pull request by q66**
This brings in a batch of template changes, including:

- We are now using python3
- Cleaned up distfiles fetching section (clearer, shorter)
- Binary distfiles now use standard install.sh scripts
- Imported various patches from Adélie Linux
- We are now using config.toml instead of configure
- Other assorted changes

TODO:

- [x] test crosscompiling
- [x] generate bootstrap binaries for stuff rust doesn't ship
- [x] get the binaries uploaded and update their checksums
- [x] test on more archs (pending @jnbr)
- [x] test firefox, etc

@jnbr

### CODE REVIEW

**Comment by q66** [comment]
@Gottox updated distfiles to upload at https://void-power.octaforge.org/distfiles/

**Comment by q66** [comment]
Also added cargo update.

**Comment by Gottox** [comment]
Done

File: srcpkgs/rust/template [hunk 1]
```diff
@@ -149,85 +130,131 @@ post_patch() {
...
+		[rust]
+		codegen-units = 1
+		debuginfo = ${_use_debug}
+		debuginfo-lines = ${_use_debug}
+		debuginfo-only-std = ${_use_debug}
+		debuginfo-tools = ${_use_debug}
+		backtrace = true
+		channel = "stable"
+		rpath = ${_use_rpath}
+		jemalloc = false
```
Comment (Cogitri) [inline srcpkgs/rust/template]: Didn't Rust remove jemalloc from rustc?

Comment (q66) [reply srcpkgs/rust/template]: It's still in the template toml file...

Comment (q66) [reply srcpkgs/rust/template]: also checked, jemalloc is still here

**Comment by jnbr** [comment]
merged: 432783a249e9a8d27164b9da9f911b095893098e

**Pull request closed without merging**